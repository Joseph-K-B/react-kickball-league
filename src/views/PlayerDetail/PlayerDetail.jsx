import { useEffect, useState } from "react"
import { Link, useParams } from "react-router-dom"
import { getPlayerById } from "../../services/players"

function PlayerDetail() {
    const{ id } = useParams()
    const[player, setPlayer] = useState(null)

    useEffect(() => {
        getPlayerById(id).then((res) => setPlayer(res))
    }, [id])

    if(!player) return <h2>Loading player info...</h2>

    return(
        <>
        <h1>{player.name}</h1>
        <Link to={`/teams/${player.teams.id}`}>
        <h3>{player.teams.name}</h3>
        </Link>
        <h3>{player.teams.city}, {player.teams.state}</h3>        
        <h3>{player.position}</h3>
        </>
    )
}
export default PlayerDetail