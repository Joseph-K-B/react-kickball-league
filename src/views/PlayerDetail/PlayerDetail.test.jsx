import { render, screen } from '@testing-library/react'
import { MemoryRouter, Route } from 'react-router-dom'
import PlayerDetail from './PlayerDetail';

it('renders team details according to id param', async () => {
    const {container} = render(
        <MemoryRouter initialEntries={['/players/1']}>
            <Route path="/players/:id">
                <PlayerDetail />
            </Route>
        </MemoryRouter>
    );

    screen.getByText('Loading player info...')

    const player = await screen.findByText(/Ben E. Jetts/, {exact: false});
    const location = await screen.findByText(/Portland, OR/, {exact: false});
    expect(player).toBeInTheDocument();
    expect(location).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})