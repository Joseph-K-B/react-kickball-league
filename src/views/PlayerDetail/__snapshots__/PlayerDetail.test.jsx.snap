// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders team details according to id param 1`] = `
<div>
  <h1>
    Ben E. Jetts
  </h1>
  <a
    class="link"
    href="/teams/1"
  >
    <h3>
      Bridge City Sneakers
    </h3>
  </a>
  <h3>
    Portland
    , 
    OR
  </h3>
  <h3>
    Pitcher
  </h3>
  <a
    class="link"
    href="/players"
  >
    <p>
      Player Directory
    </p>
  </a>
</div>
`;
